routing/router.go

```diff
@@ -1101,3 +1104,54 @@ func (r *ChannelRouter) AddProof(chanID lnwire.ShortChannelID,
 	info.AuthProof = proof
 	return r.cfg.Graph.UpdateChannelEdge(info)
 }
+
+// getGraphDiameter is a function that determines the maximum distance between
+// two nodes in the graph, where the distance is calculated as the minimum number
+// of hops from src to dest
+func (r *ChannelRouter) getGraphDiameter() (uint8,error) {
+	// local variable to represent the graph diameter
+	var diam uint8
+	// If Cache is non-empty, then return Cached value
+	if r.diamCache != 0 {
+		// Acquire Mutex to read Cached value
+		r.diamCacheMtx.Lock()
+		diam = r.diamCache
+		r.diamCacheMtx.Unlock()
+		return diam
+	}
+	// x,y used to pass to findPath
+	x := make(map[vertex]struct{})
+	y := make(map[uint64]struct{})
+
+	if err := r.cfg.Graph.ForEachNode(nil, function (tx *bolt.Tx, src *channeldb.LightningNode) error {
+		var diamPath []*ChannelHop
+		if err :=  r.cfg.Graph.ForEachNode(tx, function (_ *bolt.Tx, dest *channeldb.LightningNode) error {
+			// Distance between two identical nodes irrelevant
+			if src == dest {
+				return nil
+			}
+			// Determine the shortest Path between src and dest
+			diamPath = findPath(r.cfg.Graph,src,dest.PubKey,x,y,1)
```

Comment:
```
func findPath(graph *channeldb.ChannelGraph, sourceNode *channeldb.LightningNode,
	target *btcec.PublicKey, ignoredNodes map[vertex]struct{},
	ignoredEdges map[uint64]struct{}, amt btcutil.Amount) ([]*ChannelHop, error) { ... }
```

It looks like you're setting `diamPath` to the tuple `([]*ChannelHop, error)` instead of unpacking just the
path `[]*ChannelHop`